test/functionalTest/cases/3119_qc_nullcharacter_payload/qc_nullcharacter_payload.test

```diff
@@ -0,0 +1,71 @@
+# Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Inappropriate parameter in query context payload.
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. Request context query with inappropriate value parameter in payload.
+#
+echo "01. Request context query with inappropriate value parameter in payload."
+echo "========================================================================"
+payload='{
+    "entities": [
+        {
+            "type": "Room",
+            "isPattern": "true",
+            "id": "\u0000"
+        }
+    ]
+}'
+orionCurl --url '/v1/queryContext' --payload "$payload"
+echo
+echo
+
+
+--REGEXPECT--
+01. Request context query with inappropriate value parameter in payload.
+========================================================================
+HTTP/1.1 200 OK
+Content-Length: 128
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "errorCode": {
+        "code": "400",
+        "details": "empty entityId:id",
```

Comment:
Details is misleading... in this case what we have is actually something like "null characters are not allowed in isPattern entity id" or a similar wordking.

Reply:
Entity ID with Null characters are actually an empty ID.

Reply:
But you could have something like entity id = `\0afternull`. Strictly speaking that is not empty ID, but it has null... or not?

Reply:
As `"\0"` is 0, the string should end there and it is empty ...

Reply:
@fgalan Your concern was fixed now. Previously we were getting empty ID error messages. After few changes which I've just commited please find the request and response below:
```
[root@localhost fiware-orion]# (curl localhost:1026/v1/queryContext -s -S --header 'Content-Type: application/json' \
>     --header 'Accept: application/json' -d @- | python -mjson.tool) <<EOF
> {
>     "entities": [
>         {
>             "type": "Room",
>             "isPattern": "true",
>             "id": "\u0000"
>         }
>     ]
> }
> EOF
{
    "errorCode": {
        "code": "400",
        "details": "invalid regex for entity id pattern",
        "reasonPhrase": "Bad Request"
    }
}
```

Reply:
Please track this at PR #3125